pull.go

```diff
@@ -62,7 +68,12 @@ func Pull(ctx context.Context, imageName string, options PullOptions) (imageID s
 	libimageOptions.OciDecryptConfig = options.OciDecryptConfig
 	libimageOptions.AllTags = options.AllTags
 	libimageOptions.RetryDelay = &options.RetryDelay
-	libimageOptions.DestinationLookupReferenceFunc = cacheLookupReferenceFunc(options.BlobDirectory, types.PreserveOriginal)
+	libimageOptions.SourceLookupReferenceFunc = options.SourceLookupReferenceFunc
+	if options.DestinationLookupReferenceFunc != nil {
+		libimageOptions.DestinationLookupReferenceFunc = options.DestinationLookupReferenceFunc
+	} else {
+		libimageOptions.DestinationLookupReferenceFunc = cacheLookupReferenceFunc(options.BlobDirectory, types.PreserveOriginal)
```

Comment:
The reference function is ignored on this path. That should either be fixed, or the combination of options should be rejected; or, at the very least, it should be clearly documented.

Same on push.

Reply:
I'm not sure I follow this; this path is only reached if `options.DestinationLookupReferenceFunc` is nil. Which combination of options do you mean?

Reply:
I’m sorry, I‘ve been really unclear.

The combination of `options.SourceLookupReferenceFunc` + `options.BlobDirectory` results in `BlobDirectory` being ignored AFAICS; and `BlobDirectory != ""`, although under-documented, is basically the opt in to using the built-in blob cache.

So, in the pull options here, push options, and `buildah/define/*`, the `BlobDirectory` option and the options added by this PR are mutually exclusive.

Reply:
Thanks, added some documentation to clarify when `BlobDirectory` is overridden.

Reply:
Document that in `PullOptions` and `PushOptions` as well, please.

Reply:
Done.